File: week_1/day-4-sliding-window/maxSumSubarraySizeK.js
const nums = [2, 1, 5, 1, 3, 2]; // INPUT
const k = 3;

function maxSumSubarray(nums, k) {
    if (k > nums.length) return null;

    let windowSum = 0;
    for (let i = 0; i < k; i++) {
        windowSum += nums[i];
    }

    let maxSum = windowSum;

    for (let i = k; i < nums.length; i++) {
        windowSum += nums[i];
        windowSum -= nums[i - k];
        maxSum = Math.max(maxSum, windowSum);
    }
    return maxSum;
}

console.log(maxSumSubarray(nums, k)); // Output: 9

File: week_1/day-7-revision/max_sum_subarray_size_k.py
nums = [2, 1, 5, 1, 3, 2]  # INPUT
k = 3

def maxSumSubarray(nums, k):
    if k > len(nums):
        return None

    window_sum = sum(nums[:k])
    max_sum = window_sum

    for i in range(k, len(nums)):
        window_sum += nums[i]
        window_sum -= nums[i - k]
        max_sum = max(max_sum, window_sum)

    return max_sum

print(maxSumSubarray(nums, k))  # Output: 9
